api/internal/postgres/postgres.go

```diff
@@ -0,0 +1,359 @@
+)
+
+
+/*
+ * Stupid helper to join together terms by some operator
+ */
+func join(op operator, terms ...*string) (*string) {
+	// Remove all nil values before combining
+	_terms := make([]string, 0)
+	for _, elem := range terms {
+		if elem != nil {
+			_terms = append(_terms, *elem)
+		}
+	}
+
+	if len(_terms) == 0 {
+		return nil
+	}
+
+	out := strings.Join(_terms, string(op))
+	return &out
+}
+
+/*
+ *  Map a manifest-instance (filter) to SQL using the jsonb containment
+ *  operator ( @> )
+ */
+func containsFilter(filter *Manifest, dbColumn string) (*string) {
+	if filter == nil {
+		return nil
+	}
+
+	b, _ := json.Marshal(filter)
+
+	contains := fmt.Sprintf("%s @> '%s'", dbColumn, string(b))
+	return &contains
+}
+
+func fmtCoordinates(coords []coordinate) (*string) {
+	if len(coords) == 0 {
+		return nil
+	}
+
+	cs := make([]string, 0)
+	for _, c := range coords {
+		// TODO utm precision
+		cs = append(cs, fmt.Sprintf("%f %f", c.X, c.Y))
+	}
+	out := strings.Join(cs, ", ")
+	return &out
+}
+
+/*
+ * Map the recursive ManifestFilter into SQL syntax.
+ */
+func jsonbFilter(
+	filter *ManifestFilter,
+	dbColumn string,
+	op operator,
+) (*string) {
+	if filter == nil {
+		return nil
+	}
+
+	reduce := func(filters *[]ManifestFilter, op operator) (*string) {
+		if filters == nil {
+			return nil
+		}
+
+		processed := make([]string, 0)
+
+		for _, filter := range *filters {
+			x := jsonbFilter(&filter, dbColumn, op)
+			if x != nil {
+				processed = append(processed, *x)
+			}
+		}
+
+		if len(processed) == 0 {
+			return nil
+		}
+
+		out := fmt.Sprintf("(%s)", strings.Join(processed, string(op)))
+		return &out
+	}
+
+	filters := make([]string, 0)
+
+	eq := containsFilter(filter.Eq, dbColumn)
+	if eq != nil {
+		filters = append(filters, *eq)
+	}
+
+	or := reduce(filter.Or, Or)
+	if or != nil {
+		filters = append(filters, *or)
+	}
+
+	and := reduce(filter.And, And)
+	if and != nil {
+		filters = append(filters, *and)
+	}
+
+	if len(filters) == 0 {
+		return nil
+	}
+
+	out := fmt.Sprintf("%s", strings.Join(filters, string(op)))
+	return &out
+}
+
+func gisPoint(p *Point) (*string) {
+	if p == nil {
+		return nil
+	}
+
+	// TODO utm precision
+	out := fmt.Sprintf("POINT(%f %f)", p.X, p.Y)
+	return &out
+}
+
+func gisLinestring(line *Linestring) (*string) {
+	if line == nil {
+		return nil
+	}
+
+	coords := fmtCoordinates(line.Coords)
+	if coords == nil {
+		return nil
+	}
+
+	out := fmt.Sprintf("LINESTRING(%s)", *coords)
+	return &out
+}
+
+func gisPolygon(poly *Polygon) (*string) {
+	if poly == nil {
+		return nil
+	}
+
+	coords := fmtCoordinates(poly.Coords)
+	if coords == nil {
+		return nil
+	}
+
+	out := fmt.Sprintf("POLYGON((%s))", *coords)
+	return &out
+}
+
+func gisIntersects(column string, geom *string) (*string) {
+	if geom == nil {
+		return nil
+	}
+	out := fmt.Sprintf("ST_Intersects(%s, '%s')", column, *geom)
+	return &out
+}
+
+/*
+ * Map a Geometry instance into one or more PostGIS ST_Intersects functions,
```

Comment:
This decision was mainly driven by how I wanted the query syntax to end up. As I state in the docs you typically only want one of them in a single query. (there is always the possibility of doing several queries and concat results by the way). We can force that only one is present, but it would also require some changes to the query syntax.